pkg/vacuum/vacuum.go

```diff
@@ -0,0 +1,234 @@
+package vacuum
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+const (
+	sqlStart            = "SELECT pg_try_advisory_lock(1982010619820106)"
+	sqlListChunks       = "SELECT format('%I.%I', schema_name, table_name) from _ps_catalog.chunks_to_vacuum_freeze"
```

Comment:
I would add `LIMIT 1000` here and, perhaps, a random `ORDER BY`, just in case.

Reply:
the view itself has an order by, but yeah the limit makes a lot of sense

Reply:
2dda9cbd732cb5c13e78e0405618fa4e9fe21063
`LIMIT 1000` and continue to process batches while we get results